bindings/python-sdk/python_sdk.py

```diff
@@ -0,0 +1,341 @@
+
+    user_config_file is a .yaml file with the following three fields -
+    host: "URL of master"
+    username: "user-name"
+    password: "pwd-of-user"
+    """
+
+    def __init__(self, user_config_file: str):
+        self.user_config_file = user_config_file
+        self.configuration = None
+        self.auth = None
+        self.checkpoints = None
+        self.cluster = None
+        self.commands = None
+        self.experiments = None
+        self.internal = None
+        self.models = None
+        self.notebooks = None
+        self.shells = None
+        self.templates = None
+        self.tensorboards = None
+        self.trials = None
+        self.users = None
+
+    def initialize(self):
+        try:
+            f = open(self.user_config_file)
+        except OSError:
+            print("Error in opening user config file: {}".format(user_config_file))
+            sys.exit(1)
+
+        user_config = _parse_config_file(f)
+
+        self.configuration = swagger_client.Configuration()
+        self.configuration.host = user_config['host']
+        self.configuration.username = user_config['username']
+        self.configuration.password = user_config['password']
+        self.configuration.api_key_prefix['Authorization'] = 'Bearer'
+
+        self.auth = swagger_client.AuthenticationApi(swagger_client.ApiClient(self.configuration))
+        api_response = self.auth.determined_login(models.V1LoginRequest(self.configuration.username, self.configuration.password))        
+        self.configuration.api_key['Authorization'] = api_response.token
+        
+        # Initialize API objects with the configurations
+
+        self.checkpoints = swagger_client.CheckpointsApi(swagger_client.ApiClient(self.configuration))
+        self.cluster = swagger_client.ClusterApi(swagger_client.ApiClient(self.configuration))
+        self.commands = swagger_client.CommandsApi(swagger_client.ApiClient(self.configuration))
+        self.experiments = swagger_client.ExperimentsApi(swagger_client.ApiClient(self.configuration))
+        self.internal = swagger_client.InternalApi(swagger_client.ApiClient(self.configuration))
+        self.models = swagger_client.ModelsApi(swagger_client.ApiClient(self.configuration))
+        self.notebooks = swagger_client.NotebooksApi(swagger_client.ApiClient(self.configuration))
+        self.shells = swagger_client.ShellsApi(swagger_client.ApiClient(self.configuration))
+        self.templates = swagger_client.TemplatesApi(swagger_client.ApiClient(self.configuration))
+        self.tensorboards = swagger_client.TensorboardsApi(swagger_client.ApiClient(self.configuration))
+        self.trials = swagger_client.TrialsApi(swagger_client.ApiClient(self.configuration))
+        self.users = swagger_client.UsersApi(swagger_client.ApiClient(self.configuration))
+
+    def activate_experiment(self, exp_id: int) -> int:
+        if self.experiments is None:
+            self.initialize()
+
+        ret_val = 0
```

Comment:
q: it looks weird to me that this function responds with 1 and quietly prints out the exception if there are exceptions. A common pattern would be to throw out our exceptions directly and let users handle them.

Besides, it looks most of the functions use one pattern for error handling. We should separate out a function to do the error handling.

Reply:
Going to refactor the function. It doesn't need to do error handling, and can have the original exceptions take care of it.